ui/hooks/useSegmentContext.js

```diff
@@ -25,22 +26,35 @@ export function useSegmentContext() {
   const txData = useSelector(txDataSelector) || {};
   const confirmTransactionOrigin = txData.origin;
 
-  const referrer = confirmTransactionOrigin
-    ? {
-        url: confirmTransactionOrigin,
-      }
-    : undefined;
+  const referrer = useMemo(
+    () =>
+      confirmTransactionOrigin
+        ? {
+            url: confirmTransactionOrigin,
+          }
+        : undefined,
+    [confirmTransactionOrigin],
+  );
 
-  const page = match
-    ? {
-        path: match.path,
-        title: PATH_NAME_MAP[match.path],
-        url: match.path,
-      }
-    : undefined;
+  const page = useMemo(
+    () =>
+      match?.path
+        ? {
+            path: match.path,
+            title: PATH_NAME_MAP[match.path],
+            url: match.path,
+          }
+        : undefined,
+    [match?.path],
+  );
 
-  return {
-    page,
-    referrer,
-  };
+  const returnResult = useMemo(
+    () => ({
+      page,
+      referrer,
+    }),
+    [page, referrer],
+  );
+
+  return returnResult;
```

Comment:
This hook returned unstable data, which causes the `trackEvent` function (we use throughout the codebase) to change.